tests/chainer_tests/functions_tests/pooling_tests/test_average_pooling_nd.py

```diff
@@ -0,0 +1,180 @@
+import unittest
+
+import functools
+import itertools
+import mock
+import numpy
+import operator
+import six
+
+import chainer
+from chainer import cuda
+from chainer import functions
+from chainer import gradient_check
+from chainer import testing
+from chainer.testing import attr
+from chainer.testing import condition
+from chainer.utils import conv
+
+
+def pooling_patches(dims, ksize, stride, pad):
```

Comment:
I agree to share the implementation of `pooling_patches` function, but would not like to make the test of the average pooling layer depend on that of the max pooling layer because of their semantics.

How about to add a file like `chainer/testing/pooling_nd.py` where to place `pooling_patches`?